src/store/PokemonsSlice.ts

```diff
@@ -27,8 +31,14 @@ const pokemonsSlice = createSlice({
     setNextPokemons: (state, action: PayloadAction<string>) => {
       state.nextPokemons = action.payload.slice(action.payload.indexOf('?'));
     },
+    setPokemonsTypes: (state, action: PayloadAction<ColorTypes[]>) => {
+      state.pokemonsTypes = action.payload.reduce((accumulator, key, index) => {
+        return { ...accumulator, [key.name]: colors[index] };
+      }, {});
```

Comment:
I wouldn't recommend structure like this. I think that if you have Pokemon type, you can get bg or any other colors for it, you can make it directly in component. No need to create separate state for this in store. And for sure, you don't need to use index for this as it too unsafe